### PR TITLE
feat: branches compare merge [CS-36487]

### PR DESCRIPTION
**Pull request by nadeem-cs**

### CODE REVIEW

File: lib/stack/branch/index.js [hunk 1]
```diff
@@ -44,6 +47,29 @@ export function Branch (http, data = {}) {
...
+    /**
+     * @description Compare allows you compare any or specific ContentType or GlobalFields.
+     * @memberof Branch
+     * @func compare
+     * @returns {Compare} Instance of Compare.
+     * @example
+     * import * as contentstack from '@contentstack/management'
+     * const client = contentstack.client()
+     *
+     * client.stack({ api_key: 'api_key'}).branch('branch_uid').compare()
```
Comment (uttamukkoji) [inline lib/stack/branch/index.js:60]: @nadeem-cs Change example to include branch in compare function `client.stack({ api_key: 'api_key'}).branch('branch_uid').compare('compare_branch')`

File: lib/stack/branch/index.js [hunk 2]
```diff
@@ -77,6 +103,64 @@ export function Branch (http, data = {}) {
...
+     * @description Merge allows user to merge branches in a Stack.
+     * @memberof Branch
+     * @func merge
+     * @returns {Object} Response Object
+     *
+     * @example
+     * import * as contentstack from '@contentstack/management'
+     * const client = contentstack.client()
+     *
+     * client.stack({ api_key: 'api_key'}).branch().merge({
```
Comment (uttamukkoji) [inline lib/stack/branch/index.js:117]: @nadeem-cs change example as per new implementation

File: types/stack/branch/compare.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Response } from '../../contentstackCollection';
```
Comment (uttamukkoji) [inline types/stack/branch/compare.d.ts:1]: @nadeem-cs add `compare`, `merge`, and `mergeQueue` in `branch.d.ts` file.

File: lib/stack/branch/compare.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import { get } from '../../entity'
+
+/**
+ *
+ * @namespace Branch
+ */
+export function Compare (http, data = {}) {
+  this.stackHeaders = data.stackHeaders
+  this.urlPath = `/compare`
```
Comment (uttamukkoji) [inline lib/stack/branch/compare.js:9]: @nadeem-cs check with Anup if need to update this url as per specification in docs. Or will this remain same as we have in dev-16.

File: test/unit/mock/objects.js [hunk 5]
```diff
@@ -587,6 +587,84 @@ const installationMock = {
   uid: 'installation_uid'
 }
 
+const globalFieldDiff = [{
+  "uid": "global_field_uid",
```
Comment (uttamukkoji) [inline test/unit/mock/objects.js:591]: @nadeem-cs Js file JSON key should not include quote ex `"uid"` and change double quote to single quote `lint-issue`

File: types/stack/branch/compare.d.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+import { AnyProperty } from '../../utility/fields';
+import { Pagination } from '../../utility/pagination';
+
+export interface Compare {
+    all(params: Pagination & AnyProperty): Promise<AnyProperty>
+    contentTypes(params: Params) : Promise<Response>
```
Comment (uttamukkoji) [inline types/stack/branch/compare.d.ts:7]: @nadeem-cs update return type from `Promise<Response>` to `Promise<AnyProperty>` for `contentTypes` and `globalFields` function

File: types/stack/branch/index.d.ts [hunk 7]
```diff
@@ -1,7 +1,12 @@
 import { AnyProperty, SystemFields } from "../../utility/fields";
 import { Queryable, SystemFunction } from "../../utility/operations";
+import { Compare } from "./compare";
+import { MergeQueue, MergeQueues } from "./merge-queue";
 
 export interface Branch extends SystemFields, SystemFunction<Branch> {
+    compare(compareBranchUid: string): Compare
+    mergeQueue(mergeJobUid?: string): MergeQueue | MergeQueues
+    merge(mergeObj: IMergeObj, params: IMergeParams)
```
Comment (uttamukkoji) [inline types/stack/branch/index.d.ts:9]: @nadeem-cs please move `mergeQueue` and `merge` function in Branches interface

File: lib/stack/branch/compare.js (hunk 4 above)
Comment (nadeem-cs) [reply lib/stack/branch/compare.js:9]: As per the latest msg by Anup, v3/stack/branch/compare will be the final route. Should I make the changes acc to that or wait until the final routing is done? @uttamukkoji

**Pull request closed without merging**